backend/api/llm_providers.py

```
@@ -0,0 +1,161 @@
+        return None
+    return hashlib.sha256(api_key.encode("utf-8")).hexdigest()[:8]
+
+async def check_admin_access(user_id: str = Depends(get_current_user)):
+    # In a real app, check role from DB or token
+    if user_id != "admin":
+        raise HTTPException(status_code=403, detail="Admin access required")
+    return user_id
+
+@router.get("", response_model=List[LLMProviderResponse])
+async def list_providers(
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)